### ISSUE TITLE
Test: Guestbook application should create and stop a working app fails on Windows

### ISSUE THREAD

**Issue by adelina-t** [kubernetes/kubernetes#80534]
Failing on: 

- aks-engine-windows-master-staging
- aks-engine-windows-master
- gce-windows-master
- flannel-*-windows-master

Which test:

[sig-cli] Kubectl client [k8s.io] Guestbook application should create and stop a working application [Conformance]

**Testgrid link**:

https://testgrid.k8s.io/sig-windows#aks-engine-azure-windows-master-staging

**Reason for failure**:
```

Jul 23 23:48:34.261: INFO: Failed to get response from guestbook. err: <nil>, response: <br />
<b>Fatal error</b>:  Uncaught Predis\Response\ServerException: DENIED Redis is running in protected mode because protected mode is enabled, no bind address was specified, no authentication password is requested to clients. In this mode connections are only accepted from the loopback interface. If you want to connect from external computers to Redis you may adopt one of the following solutions: 1) Just disable protected mode sending the command 'CONFIG SET protected-mode no' from the loopback interface by connecting to Redis from the same host the server is running, however MAKE SURE Redis is not publicly accessible from internet if you do so. Use CONFIG REWRITE to make this change permanent. 2) Alternatively you can just disable the protected mode by editing the Redis configuration file, and setting the protected mode option to 'no', and then restarting the server. 3) If you started the server manually just for testing, restart it with the '--protected-mode no' option. 4) Setup a bind address or an authentication password. NOTE:  in <b>C:\Users\ContainerAdministrator\AppData\Roaming\Apache24\htdocs\Predis\src\Client.php</b> on line <b>370</b><br />
```

**Anything else we need to know**:

Most probably related to the recent transition to redis:5.0.5. Windows still uses a Redis port that is version 3.2 . Since there is no Windows version newer than 3.2 for windows we need to investigate how to accommodate the test within the existing image.

**Comment by benmoss** [comment]
I am seeing `Kubectl logs should be able to retrieve and filter logs [Conformance]` fail also due to https://github.com/kubernetes/kubernetes/pull/80126

It looks like the message that Redis prints out has changed

**Comment by benmoss** [comment]
Ah I see that there's a separate issue for that https://github.com/kubernetes/kubernetes/issues/80265

**Comment by PatrickLang** [comment]
After discussions with SIG-CLI and SIG-Apps they're ok with the approach of simplifying this down to something that works with agnhost. SIG-Apps asked to double check that service ip/port variables are covered elsewhere first, then tag them for review.

https://docs.google.com/document/d/1r0YElcXt6G5mOWxwZiXgGu_X6he3F--wKwg-9UBc29I/edit# - See Aug 14
https://docs.google.com/document/d/1LZLBGW2wRDwAfdBNHJjFfk9CFoyZPcIYGWU7R1PQ3ng/edit# - see Sep 9

**Comment by alejandrox1** [comment]
related to https://testgrid.k8s.io/sig-release-master-informing#gce-windows-master
/priority critical-urgent
/milestone v1.17

**Comment by bclau** [comment]
> 
> 
> @bclau @PatrickLang any update on this? It's still consistently failing on our release jobs

Same as before: The reason for failure is that the agnhost image version was bumped, but it was not built and published yet.

We're waiting on the kubernetes infra team to setup the automatic image building / publishing process for the kubernetes E2E test images.

**Comment by droslean** [comment]
/reopen

Reopening since the job is not green. @bclau @listx

**Comment by droslean** [comment]
https://github.com/kubernetes/kubernetes/pull/84046

**Comment by droslean** [comment]
/reopen

#84046 Doesn't resolve the issue.

@dims @listx @bclau ^^^^

**Comment by PatrickLang** [comment]
@bclau 

Seeing this on GCP runs. This is still skipped on Azure. https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce-1-17/1197156437944438786

```
Nov 20 14:51:59.288: INFO: Failed to get response from guestbook. err: the server responded with the status code 417 but did not return more information (get services frontend), response: 
Nov 20 14:52:09.340: INFO: Failed to get response from guestbook. err: the server responded with the status code 417 but did not return more information (get services frontend), response: 
Nov 20 14:52:14.341: FAIL: Cannot added new entry in 180 seconds.
Full Stack Trace
k8s.io/kubernetes/test/e2e/kubectl.validateGuestbookApp(0x53e7da0, 0xc00133d4a0, 0xc002bfaa60, 0xc)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl/kubectl.go:2315 +0x551
k8s.io/kubernetes/test/e2e/kubectl.glob..func2.7.2()
skipped 227 lines unfold_more
Latency metrics for node e2e-1b6951d679-6c91a-windows-node-group-zgvw
Nov 20 14:52:32.972: INFO: Waiting up to 3m0s for all (but 3) nodes to be ready
STEP: Destroying namespace "kubectl-7653" for this suite.

```

**Comment by alejandrox1** [comment]
Similar failures in master-informing
```
Nov 22 06:06:23.858: INFO: Running '/home/prow/go/src/k8s.io/gce-k8s-windows-testing/kubernetes/platforms/linux/amd64/kubectl --server=https://34.83.46.157 --kubeconfig=/workspace/.kube/config create -f - --namespace=kubectl-97'
Nov 22 06:06:24.212: INFO: stderr: ""
Nov 22 06:06:24.212: INFO: stdout: "deployment.apps/agnhost-slave created\n"
STEP: validating guestbook app
Nov 22 06:06:24.212: INFO: Waiting for all frontend pods to be Running.
Nov 22 06:16:24.462: FAIL: Unexpected error:
    <*errors.errorString | 0xc0007161f0>: {
        s: "Timeout while waiting for pods with labels \"app=guestbook,tier=frontend\" to be running",
    }
    Timeout while waiting for pods with labels "app=guestbook,tier=frontend" to be running
occurred
```
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce/1197741444253618176#1:build-log.txt%3A23104

**Comment by dims** [comment]
@PatrickLang @bclau i don't see this as something that needs to block 1.17.0 release. agree/disagree?

**Comment by liggitt** [comment]
is a test-only issue actually critical-urgent if it has been open for 6 months and has not been release-blocking?

**Comment by PatrickLang** [comment]
This has been a buggy test that's taken multiple iterations of refactoring to fix. We haven't been sitting still on this, but it's been hard to get reviewers to merge everything. 

@spiffxp  - are you working on driving down the flakes? this one is actionable with PRs ready to merge.

I'll change the priority since we've shipped a release without fixing this test issue.
/priority important-soon

**Comment by PatrickLang** [comment]
/remove-priority critical-urgent

This was marked critical-urgent as we were trying to get it into the 1.17 review queue. Dropping this back to important-soon
